api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -704,15 +712,18 @@ private FirecloudWorkspaceId getFcBillingProject(DbAccessTier accessTier, Worksp
                               || WorkspaceAccessLevel.WRITER.equals(u.getRole()))
                           && !finalWorkflowUsers.contains(u.getEmail())
                           && !u.getEmail().equals(userProvider.get().getUsername()))
-              .map(u -> u.getEmail())
+              .map(UserRole::getEmail)
               .collect(Collectors.toList());
-      iamService.revokeWorkflowRunnerRoleForUsers(
-          dbWorkspace.getGoogleProject(), userLostPermission);
+      List<String> failedRevocations =
+          iamService.revokeWorkflowRunnerRoleForUsers(
+              dbWorkspace.getGoogleProject(), userLostPermission);
+      if (!failedRevocations.isEmpty()) {
+        resp.setFailedWorkflowRevocations(failedRevocations);
+      }
     }
 
     // This automatically enforces the "canShare" permission.
```

Comment:
Unrelated, existing issue: re-reviewing this code, I believe this is a bug: https://precisionmedicineinitiative.atlassian.net/browse/RW-8158